File: src/PHP/BehavioralPatterns/Mediator/CommunicationHub/Participants/Participant.php
<?php

namespace BehavioralPatterns\Mediator\CommunicationHub\Participants;

use BehavioralPatterns\Mediator\CommunicationHub\Mediators\IMediator;

abstract class Participant
{
    public IMediator $mediator;

    public function setMediator($mediator): void
    {
        $this->mediator = $mediator;
    }

    abstract public function receive($sender, $args);
}

File: src/Java/BehavioralPatterns/Mediator/CommunicationHub/src/Participants/Participant.java
package Participants;

import Mediators.IMediator;

public abstract class Participant {
    protected IMediator mediator;

    public IMediator getMediator() {
        return mediator;
    }

    public void setMediator(IMediator mediator) {
        this.mediator = mediator;
    }

    public abstract void receive(Participant sender, Object args);
}

